src/libs/elektra/CMakeLists.txt

```diff
@@ -46,6 +46,10 @@ add_custom_command (OUTPUT exported_symbols.c
 include (LibAddMacros)
 add_headers (HDR_FILES)
 
+if (CMAKE_C_COMPILER_ID STREQUAL "GNU")
+	set_source_files_properties (keyset.c PROPERTIES COMPILE_FLAGS "-Wno-stringop-truncation")
```

Comment:
> Is it a good idea to silence the warnings?

If there is no reason why we cut the output one character before `strncpy` is able to fill in the terminating `\0` character, then we can also fix the warnings by increasing the size parameter by one. Another option would be to use `memcpy` instead of `strncpy`.

> Shouldn't we at least open an issue or an TODO for that?

I opened pull request #2288, which should fix the problem.